Classes/Controllers/PBHistorySearchMode.h

```diff
@@ -0,0 +1,25 @@
+//
+//  PBHistorySearchMode.h
+//  GitX
+//
+//  Created by Sven-S. Porst on 2016-12-14
+//
+
+#import <Foundation/NSObjCRuntime.h>
+
+#ifndef PBHistorySearchMode_h
+#define PBHistorySearchMode_h
+
+typedef enum historySearchModes {
+	kGitXBasicSeachMode = 1,
+	kGitXPickaxeSearchMode,
+	kGitXRegexSearchMode,
+	kGitXPathSearchMode,
+	kGitXMaxSearchMode    // always keep this item last
+} PBHistorySearchMode;
+
+
+PBHistorySearchMode searchModeForInteger(NSInteger modeInteger);
```

Comment:
Can you move that in PBHistorySearchController.h/m instead of creating a separate file ?

Also, `PBSearchModeForInteger`. Gotta love prefixes 😉.